server/routes/product.js

```diff
@@ -0,0 +1,158 @@
+const express = require("express");
+const router = express.Router();
+const { body, validationResult } = require("express-validator");
+const mongoose = require("mongoose");
+
+const User = require("../models/User");
+const Product = require("../models/Product");
+
+// @route     POST api/product
+// @desc      Register a product
+// @access    Public
+router.post(
+    "/",
+    [
+        body("title", "Please add title").not().isEmpty(),
+        body("price", "Please enter price").not().isEmpty(),
+        body("user_id", "Please enter user_id").not().isEmpty()
+    ],
+    async (req, res) => {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+            return res.status(400).json({ errors: errors.array() });
+        }
+
+        const { title, description, price, user_id, photos } = req.body;
+
+        try {
+            let user = await User.findById(user_id);
+            if (!user) {
+                return res.status(400).json({ msg: "User doesn't exist" });
+            }
+
+            let product = await Product.findOne({ title: title, user_id: user._id });
+
+            if (product) {
+                return res.status(400).json({ msg: "Product already exists" });
+            }
+
+
+            product = new Product({
+                title,
+                description,
+                price,
+                user_id: user._id,
+                photos,
+            });
+
+            await product.save();
+
+            
+
+            res.json({ id: product.id });
+
+        } catch (err) {
+            console.error(err.message);
+            res.status(500).send("Server Error");
+        }
+    }
+);
+
+// @route     POST api/product/:id
+// @desc      Update a product's information
+// @access    Public
+router.post(
+    "/:id", async (req, res) => {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+            return res.status(400).json({ errors: errors.array() });
+        }
+
+        const { title, description, price, user_id, photos } = req.body;
+
+        try {
+            let product_id = req.params.id;
+
+            if (!mongoose.Types.ObjectId.isValid(product_id)){
+                return res.status(400).json({ msg: "product_id not valid" });
+            }
+
+            let product = await Product.findById(product_id);
+
+            if (!product) {
+                return res.status(400).json({ msg: "Product doesn't exist" });
+            }
+
+            let user = await User.findById(product.user_id);
+
+            if (title) {product.title = title;}
```

Comment:
```suggestion
            product.title = title || "";
```

this may be a little more of a shorthand for these assignments - if it isn't null, it would be okay to assign it an empty string.

Same for other variables

Reply:
This is the update product function. The if statements checks whether the request wants the title to change. Else the title stays the same.

So having default values won't make much sense here